### PR TITLE
Adding pods to DeletePods and MarkPodsReady methods parameters

### PR DESCRIPTION
**Pull request by krzysied**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Passing pods a parameter of DeletePods and MarkPodsReady methods. Allows better flexibility regarding choosing pods listing approach.
Setup for #81167

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go [hunk 1]
```diff
@@ -91,7 +91,11 @@ func (nc *nodeLifecycleController) doEviction(fakeNodeHandler *testutil.FakeNode
 	for _, zone := range zones {
 		nc.zonePodEvictor[zone].Try(func(value scheduler.TimedValue) (bool, time.Duration) {
 			uid, _ := value.UID.(string)
-			nodeutil.DeletePods(fakeNodeHandler, nc.recorder, value.Value, uid, nc.daemonSetStore)
+			pods, err := listPodsFromNode(nc.kubeClient, value.Value)
```
Comment (wojtek-t) [inline pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: fakeNodeHandler implements clientset.Interface

Either change here to use fakeNodeHandler here or use nc.kubeClient both here and below. I guess the former seem better.

Comment (wojtek-t) [reply pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: Same below

Comment (krzysied) [reply pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: Done.

File: pkg/controller/util/node/controller_utils.go [hunk 2]
```diff
@@ -44,32 +44,26 @@ import (
 // DeletePods will delete all pods from master running on given node,
 // and return true if any pods were deleted, or were found pending
 // deletion.
-func DeletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore appsv1listers.DaemonSetLister) (bool, error) {
+func DeletePods(kubeClient clientset.Interface, pods []v1.Pod, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore appsv1listers.DaemonSetLister) (bool, error) {
```
[the remark this answers is not in the document]
Comment (krzysied) [reply pkg/controller/util/node/controller_utils.go]: Done.

**Pull request merged**